File: system-tests/selenium-java/src/test/java/com/qatraining/CartTest.java
package com.qatraining;

import org.junit.jupiter.api.*;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.NoSuchElementException;

import java.time.Duration;

import static org.junit.jupiter.api.Assertions.*;

public class CartTest {

	
    WebDriver driver;
    private static final String BASE_URL = "https://www.saucedemo.com";

    @BeforeEach
	void setUp() {
		driver = DriverFactory.createDriver();

		driver.get(BASE_URL);
		driver.findElement(By.id("user-name")).sendKeys("standard_user");
		driver.findElement(By.id("password")).sendKeys("secret_sauce");
		driver.findElement(By.id("login-button")).click();
	}

    @AfterEach
    void tearDown() {
        driver.quit();
    }

    private boolean cartBadgeExists() {
        try {
            driver.findElement(By.className("shopping_cart_badge"));
            return true;
        } catch (NoSuchElementException e) {
            return false;
        }
    }

    @Test
    void agregarProductoActualizaContador() {
        assertFalse(cartBadgeExists());

        driver.findElement(By.cssSelector("[data-test='add-to-cart-sauce-labs-backpack']")).click();

        String count = driver.findElement(By.className("shopping_cart_badge")).getText();
        assertEquals("1", count);
    }

    @Test
    void agregarMultiplesProductosSumaCorrectamente() {
        driver.findElement(By.cssSelector("[data-test='add-to-cart-sauce-labs-backpack']")).click();
        driver.findElement(By.cssSelector("[data-test='add-to-cart-sauce-labs-bike-light']")).click();

        String count = driver.findElement(By.className("shopping_cart_badge")).getText();
        assertEquals("2", count);
    }

    @Test
    void removerProductoActualizaContador() {
        driver.findElement(By.cssSelector("[data-test='add-to-cart-sauce-labs-backpack']")).click();
        driver.findElement(By.cssSelector("[data-test='remove-sauce-labs-backpack']")).click();

        assertFalse(cartBadgeExists());
    }
}

File: system-tests/selenium-js/cart.test.js
const { By } = require('selenium-webdriver');

const BASE_URL = 'https://www.saucedemo.com';

const { buildDriver } = require('./driverFactory');

async function cartBadgeExists(driver) {
  const badges = await driver.findElements(By.className('shopping_cart_badge'));
  return badges.length > 0;
}

describe('Shopping Cart', () => {
  let driver;

  beforeEach(async () => {
    driver = await buildDriver();
    await driver.manage().setTimeouts({ implicit: 5000 });

    await driver.get(BASE_URL);
    await driver.findElement(By.id('user-name')).sendKeys('standard_user');
    await driver.findElement(By.id('password')).sendKeys('secret_sauce');
    await driver.findElement(By.id('login-button')).click();
  });

  afterEach(async () => {
    await driver.quit();
  });

  test('agregar producto actualiza el contador', async () => {
    expect(await cartBadgeExists(driver)).toBe(false);

    await driver.findElement(By.css("[data-test='add-to-cart-sauce-labs-backpack']")).click();

    const badge = await driver.findElement(By.className('shopping_cart_badge')).getText();
    expect(badge).toBe('1');
  }, 15000);

  test('agregar múltiples productos suma correctamente', async () => {
    await driver.findElement(By.css("[data-test='add-to-cart-sauce-labs-backpack']")).click();
    await driver.findElement(By.css("[data-test='add-to-cart-sauce-labs-bike-light']")).click();

    const badge = await driver.findElement(By.className('shopping_cart_badge')).getText();
    expect(badge).toBe('2');
  }, 15000);

  test('remover producto actualiza el contador', async () => {
    await driver.findElement(By.css("[data-test='add-to-cart-sauce-labs-backpack']")).click();
    await driver.findElement(By.css("[data-test='remove-sauce-labs-backpack']")).click();

    expect(await cartBadgeExists(driver)).toBe(false);
  }, 15000);
});